### PR TITLE
add new arabic courses

### PR DESCRIPTION
**Pull request by AhmedElTabarani**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-ar.md [hunk 1]
```diff
@@ -40,15 +40,18 @@
...
 * [CS Master - Level 2- Dive into the Computer كيف يعمل الكمبيوتر](https://www.youtube.com/playlist?list=PLL2zWZTDFZziX_xS2bbGfLAOHVmlzURhF) - KMR Script
 * [CS50 in Arabic&rlm;](https://www.youtube.com/playlist?list=PLL2zWZTDFZzibJ49gBM2owqCzda8meSNj) - KMR Script
 * [CS50 In Arabic&rlm;](https://www.youtube.com/playlist?list=PLnrlZUDQofUv7JE4QIahAyztrQU9bnJmd) - Coders Camp
 * [Cs50 In Arabic 2022 \| كورس cs50 بالعربي كامل](https://www.youtube.com/playlist?list=PLknwEmKsW8OvsdJ64v5YljHNtt100kN6w) - Abdelrahman Gamal&rlm; (:construction: *in process*)
 * [Data Structure&rlm;](https://www.youtube.com/playlist?list=PLwCMLs3sjOY4UQq4vXgGPwGLVX1Y5faaS) - Hard Code
+* [Data Structure : JavaScript (leetcode problem solving)&rlm;](https://www.youtube.com/playlist?list=PLS-MrzRLZtmdIHJ-Osvv_sJO1Msc4VM_7) - Shadow Coding
 * [Data Structure C++&lrm;&rlm;](https://www.youtube.com/playlist?list=PLsGJzJ8SQXTcsXRVviurGei0lf_t_I4D8) - Mega Code
 * [Data Structures Full Course In Arabic&rlm;](https://www.youtube.com/playlist?list=PLCInYL3l2AajqOUW_2SwjWeMwf4vL4RSp) - Adel Nasim
+* [Grokking Algorithms](https://www.youtube.com/playlist?list=PLWCr3KJclfzGV82gewnuAoKFBt_f9FvNa) - Programming Books
```
Comment (LuigiImVector) [inline courses/free-courses-ar.md:54]:
```suggestion
* [Grokking Algorithms&rlm;](https://www.youtube.com/playlist?list=PLWCr3KJclfzGV82gewnuAoKFBt_f9FvNa) - Programming Books
```

File: courses/free-courses-ar.md [hunk 2]
```diff
@@ -113,6 +117,7 @@
 * [CS Master - Level 3 - Databases قواعد البيانات](https://www.youtube.com/playlist?list=PLL2zWZTDFZzhXQ1bcYlO3PtN4MsLiG-gy) - KMR Script
 * [Database 1 - المقرر النظرى - Fundamentals of Database Systems](https://www.youtube.com/playlist?list=PL37D52B7714788190) - محمد الدسوقى
 * [Database Design](https://www.youtube.com/playlist?list=PLkzDzmo9y3VHDFKp7LuXd-FwbefvTL5o0) - تخاريف مبرمج
+* [Designing Data Intensive Applications](https://www.youtube.com/playlist?list=PLTRDUPO2OmIljJwE9XMYE_XEgEIWZDCuQ) - Ahmed Elemam
```
Comment (LuigiImVector) [inline courses/free-courses-ar.md:120]:
```suggestion
* [Designing Data Intensive Applications&rlm;](https://www.youtube.com/playlist?list=PLTRDUPO2OmIljJwE9XMYE_XEgEIWZDCuQ) - Ahmed Elemam
```

File: courses/free-courses-ar.md [hunk 3]
```diff
@@ -170,6 +175,7 @@
 * [Learn SASS In Arabic 2021&rlm;](https://www.youtube.com/playlist?list=PLDoPjvoNmBAzlpyFHOaB3b-eubmF0TAV2) - Elzero Web School
 * [Sass&rlm;](https://www.youtube.com/playlist?list=PLDoPjvoNmBAz6bF7qObm2a1mLN_WHAWQo) - Elzero Web School
 * [Sass - Create a website&rlm;](https://www.youtube.com/playlist?list=PLDoPjvoNmBAz9sluuyOWPifXvySgrGma8) - Elzero Web School
+* [Yonisfy CS94's | Web Programming with JavaScript](https://www.youtube.com/playlist?list=PLLWuK602vNiXqkUqcoKrVjCyOrZh7Tw0e) - Mohammed Elzanaty
```
Comment (LuigiImVector) [inline courses/free-courses-ar.md:178]:
```suggestion
* [Yonisfy CS94's | Web Programming with JavaScript&rlm;](https://www.youtube.com/playlist?list=PLLWuK602vNiXqkUqcoKrVjCyOrZh7Tw0e) - Mohammed Elzanaty
```

File: courses/free-courses-ar.md [hunk 4]
```diff
@@ -189,13 +195,17 @@
...
-* [كورس جافا سكريبت كامل | Javascript Tutorial&rlm;](https://www.youtube.com/playlist?list=PLknwEmKsW8OuTqUDaFRBiAViDZ5uI3VcE) - Abdelrahman Gamal
+* [سلسلة دروس جافا سكريبت](https://www.youtube.com/playlist?list=PLYyqC4bNbCIeLEjcSPO61bsGPKEvYceb0) - أكاديمية ترميز
+* [كورس جافا سكريبت كامل \| Javascript Tutorial&rlm;](https://www.youtube.com/playlist?list=PLknwEmKsW8OuTqUDaFRBiAViDZ5uI3VcE) - Abdelrahman Gamal
 * [كورس Ajax بالعربي-ITI&rlm;](https://www.youtube.com/playlist?list=PLzCpl3aBwaY7wv9ZZkhH7lZELpz_fP81N) - Mobarmg
 * [كورس ES6 بالعربي-ITI&rlm;](https://www.youtube.com/playlist?list=PLzCpl3aBwaY5nQxWH0HaNibR9UXKICWxz) - Mobarmg
 * [Arabic JavaScript&rlm;](https://www.youtube.com/playlist?list=PLL2zWZTDFZzgU2x6Kk6w0qx5piLgMODsm) - KMR Script
 * [Arabic JavaScript ES6 الاصدار السادس من الجافاسكربت](https://www.youtube.com/playlist?list=PLL2zWZTDFZzilx_LJ_mCRDETtDOyBg0UT) - KMR Script
+* [Complete Intro to Javascript&rlm;](https://www.youtube.com/playlist?list=PLLWuK602vNiU-kIpNuw5Z7HRbV4pBaHlL) - Mohammed Elzanaty
+* [Design Patterns \| javascript \| [Arabic]&rlm;](https://www.youtube.com/playlist?list=PLS-MrzRLZtmduTfp_bReagQKg7I-GVr-5) - Shadow Coding
```
Comment (LuigiImVector) [inline courses/free-courses-ar.md:205]:
```suggestion
* [Design Patterns \| javascript \| \[Arabic\]&rlm;](https://www.youtube.com/playlist?list=PLS-MrzRLZtmduTfp_bReagQKg7I-GVr-5) - Shadow Coding
```

File: courses/free-courses-ar.md [hunk 5]
```diff
@@ -336,6 +350,8 @@
 * [Clean Code Book - بالعربي](https://www.youtube.com/playlist?list=PLwWuxCLlF_ufTMlXoJlQvImqz9wIfcWsX) - Omar Ahmed
 * [Creational Design Patterns بالعربى](https://www.youtube.com/playlist?list=PLnqAlQ9hFYdewk9UKGBcHLulZNUBpNSKJ) - Mohammed Reda
 * [Declarative Programming](https://www.youtube.com/playlist?list=PLpbZuj8hP-I6F-Zj1Ay8nQ1rMnmFnlK2f) - درة الاكواد لابن حماد
+* [Head First Object-Oriented Analysis and Design&rlm;](https://www.youtube.com/playlist?list=PLWCr3KJclfzGkG7CXBmSdlIcIpjxo4FbF) - Programming Books
+* [Software Design Patterns](https://www.youtube.com/playlist?app=desktop&list=PLrwRNJX9gLs3oQyBoXtYimY7M5aSF0_oC) - محمد يحيى
```
Comment (LuigiImVector) [inline courses/free-courses-ar.md:354]:
```suggestion
* [Software Design Patterns&rlm;](https://www.youtube.com/playlist?app=desktop&list=PLrwRNJX9gLs3oQyBoXtYimY7M5aSF0_oC) - محمد يحيى
```

**Review by LuigiImVector — COMMENTED** [review]
Missed some `&rlm;`

Also remember that atomic commits are prefered:
https://github.com/EbookFoundation/free-programming-books/blob/88c23008395da983bf824f65a68ef394ae992b89/docs/CONTRIBUTING.md#L55

File: courses/free-courses-ar.md (hunk 4 above)
Comment (AhmedElTabarani) [reply courses/free-courses-ar.md:205]: The square brackets must be escaped also ?

File: courses/free-courses-ar.md (hunk 5 above)
Comment (AhmedElTabarani) [reply courses/free-courses-ar.md:354]: No need to add `&rlm;` here 
the author name in Arabic, so the context will not swap

Comment (AhmedElTabarani) [reply courses/free-courses-ar.md:354]: image

Comment (LuigiImVector) [reply courses/free-courses-ar.md:354]: I didn't know that, could you add it to #6964?

File: courses/free-courses-ar.md (hunk 4 above)
Comment (LuigiImVector) [reply courses/free-courses-ar.md]: could be confused with those in the link

File: courses/free-courses-ar.md (hunk 5 above)
Comment (AhmedElTabarani) [reply courses/free-courses-ar.md:354]: #6964 ? You miss the issue?

Comment (AhmedElTabarani) [reply courses/free-courses-ar.md:354]: If you mean this #6714, this only for problem

`* [name in english](url) - name in arabic`
That not cause any problem

**Comment by AhmedElTabarani** [comment]
I will delete `Head First Object-Oriented Analysis and Design`, `Grokking Algorithms`, `Designing Data Intensive Applications`

Because it does courses to explain paid books as @eshellman  said here [#6764 ](https://github.com/EbookFoundation/free-programming-books/issues/6764#issuecomment-1194014907)

File: courses/free-courses-ar.md (hunk 5 above)
Comment (LuigiImVector) [reply courses/free-courses-ar.md:354]: > If you mean this #6714

ops, i confused id


> That not cause any problem

maybe you could add it just for clarification, in an "extra case 2"

**Comment by AhmedElTabarani** [comment]
@eshellman, relying on a paid book to explain is illegal? Or explain the book itself is illegal?

Because most of the courses depend on paid books such as `clean code` courses

File: courses/free-courses-ar.md (hunk 5 above)
Comment (AhmedElTabarani) [reply courses/free-courses-ar.md:354]: > maybe you could add it just for clarification, in an "extra case 2"

done

**Comment by eshellman** [comment]
What do you think - Is it a free course if the book isn't free? Many courses are taught so they don't *really* need the book, but maybe these really rely heavily on the not-free book? The test is: does the course stand on its own? We have to rely on you to make the judgement.

**Review by LuigiImVector — APPROVED** [review]

File: courses/free-courses-ar.md [hunk 6]
```diff
@@ -170,6 +173,7 @@
 * [Learn SASS In Arabic 2021&rlm;](https://www.youtube.com/playlist?list=PLDoPjvoNmBAzlpyFHOaB3b-eubmF0TAV2) - Elzero Web School
 * [Sass&rlm;](https://www.youtube.com/playlist?list=PLDoPjvoNmBAz6bF7qObm2a1mLN_WHAWQo) - Elzero Web School
 * [Sass - Create a website&rlm;](https://www.youtube.com/playlist?list=PLDoPjvoNmBAz9sluuyOWPifXvySgrGma8) - Elzero Web School
+* [Yonisfy CS94's | Web Programming with JavaScript&rlm;](https://www.youtube.com/playlist?list=PLLWuK602vNiXqkUqcoKrVjCyOrZh7Tw0e) - Mohammed Elzanaty
```
Comment (AhmedElTabarani) [inline courses/free-courses-ar.md:176]: I forget escape `|`, we need a regex to check these mistakes `if a '|' have an escape character '\' or not, or something like this.

Comment (LuigiImVector) [reply courses/free-courses-ar.md:176]: This issue is discussed on #5176

**Pull request merged**